### PR TITLE
GoLang Signing Service Challenge

### PR DESCRIPTION
**Pull request by auxdevelopment**

### CODE REVIEW

File: signing-service-challenge/README.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+This special string will be signed (`Signer.sign(secured_data_to_be_signed)`) and the resulting signature (`base64` encoded) will be returned to the client. A signature response could look like this:
+
+```json
+{ 
+    "signature": <signature_base64_encoded>,
+    "signed_data": "<signature_counter>_<data_to_be_signed>_<last_signature_base64_encoded>"
+}
+```
+
+After the signature has been created, its value shall be increased by one (`signature_counter += 1`).
```
Comment (StefanKohlbacher) [inline signing-service-challenge/README.md:55]: ... the signature_counter shall be increased ...

File: signing-service-challenge/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+#### Technical Constraints & Considerations
+
+- The system will be used by many concurrent clients accessing the same resources.
+- The `signature_counter` shall be strictly monotonically increasing and ideally without any gaps.
+- The system currently only supports `RSA` and `ECDSA` as signature algorithms. Try to design the signing mechanism in a way that allows easy extension to other algorithms without changing the core domain logic.
+- For now it is enough to store signature devices in memory. Efficiency is not a priority for this. We might want to scale out though, therefore keep in mind that we might want to switch to a relational database at some point when designing your storage logic.
+
+#### Credits
+
+This challenge is heavily influenced by the KassenSichV (Germany) as well as the RKSV (Austria) and our solutions for them.
```
Comment (StefanKohlbacher) [inline signing-service-challenge/README.md:81]: that's an interesting hint

File: signing-service-challenge/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (auxdevelopment) [reply signing-service-challenge/README.md:55]: Yes, `signature_counter` was meant. I agree, it's not worded nicely.

File: signing-service-challenge/README.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# Signature Service - Coding Challenge
+
+## Instructions
+
+Welcome challenger!  
+When you see this, you have successfully passed the first interview at fiskaly. Now
+comes the time to prove yourselves. We have the following challenge prepared for you,
```
[the remark this answers is not in the document]
Comment (auxdevelopment) [reply signing-service-challenge/README.md:7]: I'm generally not a huge fan of that intro tbh ...

**Pull request merged**